editor/directory_create_dialog.cpp

```diff
@@ -51,13 +51,17 @@ String DirectoryCreateDialog::_validate_path(const String &p_path) const {
 		return TTR("Folder name cannot be empty.");
 	}
 
+	if (p_path.contains("\\") || p_path.contains(":") || p_path.contains("*") ||
+			p_path.contains("|") || p_path.contains(">") || p_path.ends_with(".") || p_path.ends_with(" ")) {
```

Comment:
Why the name can't end with `.`?
Also `" "` is irrelevant, because the dialog does `strip_edges()` before creating the folder.

Reply:
This isn't my code afaik, it's because it hasn't been rebased in a bit.

I'll rebase and edit in an hour or two.

Reply:
Inside the loop it checks parts separated with `/`, e.g. `path/to/directory/somewhere`. It makes sense that the middle elements shouldn't have dots/spaces.

Reply:
It doesn't do anything different though, because inside the loop it still used p_path. Unless there's some weird in-place stuff I'm unaware of, seems to me it does the exact same thing (it doesn't use part)

Reply:
Well, that looks like a bug 🙃

Reply:
I'll fix it so that each part can't end with a dot or a space?

Reply:
Not sure about dot tbh, as it's valid to use, but space shouldn't be allowed.

Reply:
Might be worth checking for a space at the beginning too.

Reply:
It now does that too